torch/nn/modules/_functions.py

```diff
@@ -67,11 +67,19 @@ def forward(self, input, weight, bias, running_mean, running_var, eps, momentum,
             # world_size * (2C + 1) -> world_size * C, world_size * C, world_size * 1
             mean_all, invstd_all, count_all = torch.split(combined, num_channels, dim=1)
 
-        # remove stats from empty inputs
-        mask = count_all.squeeze(-1) >= 1
-        count_all = count_all[mask]
-        mean_all = mean_all[mask]
-        invstd_all = invstd_all[mask]
+        if not torch.cuda.is_current_stream_capturing():
+            # The lines below force a synchronization between CUDA and CPU, because
+            # the shape of the result count_all depends on the values in mask tensor.
+            # Such synchronizations break CUDA Graph capturing.
+            # See https://github.com/pytorch/pytorch/issues/78549
+            # FIXME: https://github.com/pytorch/pytorch/issues/78656 describes
+            # a better longer-term solution.
+
+            # remove stats from empty inputs
```

Comment:
Does that mean that when capturing, this will return wrong results?
Also not sure why is the `count_all[mask]` actually forcing a sync?

Reply:
> Does that mean that when capturing, this will return wrong results?

Yes. We can change that to if capturing and if `input.numel()`, error out if that's better. 

> Also not sure why is the count_all[mask] actually forcing a sync?

Because the shape of the return value depends on the values in mask?
<img width="702" alt="Screen Shot 2022-06-01 at 5 08 14 PM" src="https://user-images.githubusercontent.com/16999635/171502106-74e3d432-bf8e-441d-8562-23903c72cdb8.png">

Reply:
Ho that is a boolean mask. Ok ok!